src/moonlink/src/storage/iceberg/rest_catalog_test.rs

```diff
@@ -1,108 +1,157 @@
+        .catalog
+        .drop_table(&table_ident)
         .await
-        .unwrap();
+        .expect("error: fail to drop the table");
 
-    // Check table existence.
-    let table_ident = guard.table.clone().unwrap();
-    assert!(catalog.table_exists(&table_ident).await.unwrap());
+    if let Some(vec) = guard.tables_idents.as_mut() {
+        vec.retain(|t| t != &table_ident);
+    }
 
-    // List tables and validate.
-    let tables = catalog.list_tables(table_ident.namespace()).await.unwrap();
-    assert_eq!(tables, vec![table_ident.clone()]);
+    assert!(!writer
+        .table_exists(&table_ident)
+        .await
+        .expect("error: fail to check whether the table exist"));
+    assert_eq!(
+        writer
+            .list_tables(&ns_ident)
+            .await
+            .expect("error: fail to list the tables"),
+        vec![]
+    );
 
-    // Load table and check schema.
-    let table = catalog.load_table(&table_ident).await.unwrap();
-    let actual_schema = table.metadata().current_schema();
-    assert_is_same_schema(actual_schema.as_ref().clone(), iceberg_schema);
+    drop(writer);
 }
 
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn test_load_table() {
-    let namespace = get_random_string();
-    let table = get_random_string();
-    let guard = RestCatalogTestGuard::new(namespace.clone(), Some(table.clone()))
+async fn test_namespace_operation() {
+    let mut guard = RestCatalogTestGuard::new()
         .await
-        .unwrap();
-    let rest_catalog_config = default_rest_catalog_config();
-    let accessor_config = default_accessor_config();
-    let catalog = RestCatalog::new(
-        rest_catalog_config,
-        accessor_config,
-        create_test_table_schema().unwrap(),
-    )
-    .await
-    .unwrap();
-    let table_ident = guard.table.clone().unwrap();
-    let result = catalog.load_table(&table_ident).await.unwrap();
-    let result_table_ident = result.identifier().clone();
-    assert_eq!(table_ident, result_table_ident);
+        .expect("error: fail to create testguard");
+    let catalog = guard.catalog.clone();
+    let writer = catalog.write().await;
+
+    // create a namespace
+    let ns_name = get_random_string();
+    let ns_ident = NamespaceIdent::new(ns_name);
+    let expected_namespace = writer
+        .catalog
+        .create_namespace(&ns_ident, HashMap::new())
+        .await
+        .expect("error: fail to create a namespace");
+    guard
+        .namespace_idents
+        .get_or_insert(Vec::new())
+        .push(ns_ident.clone());
+
+    assert_eq!(
+        writer
+            .catalog
+            .get_namespace(&ns_ident)
+            .await
+            .expect("error: fail to get namespace"),
+        expected_namespace
+    );
+    assert!(writer
+        .catalog
+        .namespace_exists(&ns_ident)
+        .await
+        .expect("error: fail to check if the namespace exist"));
+
+    assert_eq!(
+        writer
+            .catalog
```

Comment:
You are listing namespaces in **root namespace**, rather than your own unique **namespace**.